src/Text/Pandoc/BCP47.hs

```diff
@@ -93,14 +93,16 @@ parseBCP47 lang =
           ds <- P.option "" (P.count 1 P.digit)
           cs <- P.many1 asciiLetter
           let var = ds ++ cs
+              lv = length var
```

Comment:
> I'd expect GHC to optimize this for us

I thought GHC in general doesn't add sharing without you telling it to, since it can be a space leak just as easily as an optimization.